src/solid-polygon-layer.ts

```diff
@@ -94,12 +98,125 @@ export class GeoArrowSolidPolygonLayer<
   static defaultProps = defaultProps;
   static layerName = "GeoArrowSolidPolygonLayer";
 
+  declare state: CompositeLayer["state"] & {
+    table: arrow.Table | null;
+    triangles: Uint32Array[] | null;
+  };
+
+  initializeState(_context: LayerContext): void {
+    this.state = {
+      table: null,
+      triangles: null,
+    };
+  }
+
+  async updateData() {
+    const { data: table } = this.props;
+    const earcutTriangles = await this._updateEarcut(table);
+    this.setState({ table: this.props.data, triangles: earcutTriangles });
+  }
+
+  async _updateEarcut(table: arrow.Table): Promise<Uint32Array[]> {
+    const polygonVector = getGeometryVector(table, EXTENSION_NAME.POLYGON);
+    if (polygonVector !== null) {
+      return this._earcutPolygonVector(polygonVector);
+    }
+
+    const MultiPolygonVector = getGeometryVector(
+      table,
+      EXTENSION_NAME.MULTIPOLYGON,
+    );
+    if (MultiPolygonVector !== null) {
+      return this._earcutMultiPolygonVector(MultiPolygonVector);
+    }
+
+    const geometryColumn = this.props.getPolygon;
+    if (ga.vector.isPolygonVector(geometryColumn)) {
+      return this._earcutPolygonVector(geometryColumn);
+    }
+
+    if (ga.vector.isMultiPolygonVector(geometryColumn)) {
+      return this._earcutMultiPolygonVector(geometryColumn);
+    }
+
+    throw new Error("geometryColumn not Polygon or MultiPolygon");
+  }
+
+  async _earcutPolygonVector(
+    geometryColumn: ga.vector.PolygonVector,
+  ): Promise<Uint32Array[]> {
+    console.log("spawning worker");
+
+    const workerTextResp = await fetch(
+      this.props.workerUrl || "http://localhost:8082/earcut-worker.js",
+    );
+    const workerText = await workerTextResp.text();
+    const worker =  BlobWorker.fromText(workerText);
+
+    const pool = Pool(
+      () => spawn(BlobWorker.fromText(workerText)),
+      8 /* optional size */,
+    );
+
+
+    // const earcutWorker = await spawn(worker);
+    console.log("spawned worker");
+    console.log(pool);
+    // console.log(earcutWorker);
+    // if (!earcutWorker) return;
+
+    const resultPromises = new Array(geometryColumn.data.length);
+    const result: Uint32Array[] = new Array(geometryColumn.data.length);
+
+    for (
+      let recordBatchIdx = 0;
+      recordBatchIdx < geometryColumn.data.length;
+      recordBatchIdx++
+    ) {
+      const polygonData = geometryColumn.data[recordBatchIdx];
+      const [preparedPolygonData, arrayBuffers] = ga.worker.preparePostMessage(
+        polygonData,
+        true,
+      );
+      pool.queue(async earcutWorker => {
+        const earcutTriangles = await earcutWorker(
+          Transfer(preparedPolygonData, arrayBuffers),
+        );
+        result[recordBatchIdx] = earcutTriangles;
+      })
+      // const earcutTriangles = await earcutWorker(
+      //   Transfer(preparedPolygonData, arrayBuffers),
+      // );
+      // result[recordBatchIdx] = earcutTriangles;
+    }
```

Comment:
iterate over all the `Data` chunks in the vector, sending each of them to the earcut threadpool and assigning the output into the result array